system/jlib/jio.ipp

```diff
@@ -99,6 +99,80 @@ protected:
         curBufferOffset += sizeGot;
         return sizeGot;
     }
+    inline bool readLineFromBuffer(bool keepCRLF, StringBuffer & outBuffer) 
+    {
+        if ((numInBuffer == curBufferOffset) && !fillBuffer())
+            return true;
+
+        bool eof = false;
+        bool foundCRLF = false;
+        while (true)
+        {
+            size32_t outLength = 0;
+            char * outStartFrom = (char *) buffer+curBufferOffset;
+            size32_t tempOffset = curBufferOffset;
+            while (tempOffset < numInBuffer)
+            {
+                if (buffer[tempOffset] == '\r') // standard case, \r\n marks a new header line.
+                {
+                    foundCRLF = true;
+                    tempOffset++;
+                    if(keepCRLF)
+                        outLength++;
+
+                    //check \n
+                    if (tempOffset < numInBuffer)
+                    {
+                        if(buffer[tempOffset] == '\n')
+                        {
+                            tempOffset++;
+                            if(keepCRLF)
+                                outLength++;
+                        }
+
+                        outBuffer.append(outLength, outStartFrom);
+                        curBufferOffset = tempOffset;
+                    }
+                    else
+                    {
+                        outBuffer.append(outLength, outStartFrom);
+                        eof = !fillBuffer();
+                        if (!eof && (buffer[0] == '\n'))
+                        {
+                            if(keepCRLF)
+                                outBuffer.append(buffer[0]);
+                            curBufferOffset++;
+                        }
+                    }
+                    break;
+                }
+                else if (buffer[tempOffset] == '\n') // deal with non-standard case, when only a \n marks a new line.
+                {
+                    foundCRLF = true;
+                    tempOffset++;
+                    if (keepCRLF)
+                        outLength++;
+
+                    outBuffer.append(outLength, outStartFrom);
+                    curBufferOffset = tempOffset;
+                    break;
+                }
+
+                outLength++;
+                tempOffset++;
+            }
+
+            if(foundCRLF)
+                break;
```

Comment:
There's a bit of inconsistency here, of when eof is returned from readLineFromBuffer.
i.e. only checked if needs to read 1 more char following \r.

It should be consistent.

Reply:
Do you want me to check a possible eof after the CRLF is found (even if it is not the case of reading 1 more char following \r)?